docs/2018-10-24-Configuration-Splitting-Simplification.md

```diff
@@ -0,0 +1,88 @@
+Configuration Splitting Simplification
+======================================
+
+Background
+----------
+
+When doing one-path reachability proofs and applying an axiom,
+`∀ Z . α(Z) → ● β(Z)`, where `●` is the strong-next symbol (aka 'next'),
+to the following reachability goal
+```
+∀ X . φ(X) → ◆ ∃ Y . ψ(X, Y)
+```
+the result will look something like
+```
+￼(∀ X . Φ'(X) → ◆ ∃ Y . ψ(X, Y))
+￼∧
+￼(∀ X . Φ(X) ∧ (¬ ∃ Z. α(Z)) → ● ◆ ∃ Y . ψ(X, Y))
+```
+where `Φ` is either `φ` or `φ` and-ed with something else.
+See `one-path-reachability-proofs.md` for details.
+
+Problem
+-------
+
+We want to rewrite `φ(X) ∧ (¬ ∃ Z. α(Z))` part of the pattern above
+to something more manageable, preferably something that does not use `not`
+and `exists`, except in cases where it can be handled by an SMT solver.
+
+We will assume that
+1. `φ(X) = t(X) ∧ p(X)` where `p(X)` is a predicate, and `t(X)` is a
+   function-like term. It may or may not unify with `α(Z)`.
```

Comment:
What is "function-like"? Is that stronger than "functional"?

Reply:
It's weaker, I added some definitions at the top.